app/[category]/v/[prefix]/[videoId]/[lang]/page.tsx

```diff
@@ -55,22 +58,35 @@ export default async function Index({params}: {params: VideoParams}) {
     );
   }
 
-  const transcriptLangCode = params.lang === 'eng' ? 'en' : params.lang;
+  const transcriptLangCode = params.lang;
 
   // Handle the fact that transcript files for english still use the 2-letter ISO-639 code.
+  const loadData = new Array<Promise<any>>;
+  loadData.push(getMetadata(params.category, params.videoId));
+  loadData.push(await DiarizedTranscript.makeFromStorage(
+      firebaseStorageAccessor, params.category, params.videoId));
+  loadData.push(await loadSpeakerControlInfo(params.category, params.videoId));
 
-  const metadata = await getMetadata(params.category, params.videoId);
-  const diarizedTranscript = await getDiarizedTranscript(params.category, params.videoId, transcriptLangCode);
-  const speakerControlInfo = await loadSpeakerControlInfo(params.category, params.videoId);
-  const subtextTranscript = transcriptLangCode === 'en' ? undefined : await getDiarizedTranscript(params.category, params.videoId, 'en');
+  const results = await Promise.all(loadData);
+
+  const metadata = results.shift();
```

Comment:
Can this block be an array destructure?

`const [metadata, diarizedTranscript, speakerControlInfo] = results`

Reply:
Nice! done.